### DISCUSSION TITLE
Rename orig.ident

### DISCUSSION THREAD

**Discussion by carloslizama** [satijalab/seurat#4873] [General]
Hey,

I got a dataset from geo where all the files was just in one matrix and I was wondering if has any easy way to regroup and rename the orig.ident.
For example All the F group regroup-rename like a control, all HD like a disease1 and all LD like disease 2.

![Screenshot from 2021-08-01 14-08-28](https://user-images.githubusercontent.com/18195075/127785969-d8067b43-0519-4afb-b4f5-74b207077153.png)

**Comment by royfrancis** [comment]
You need to have a new metadata column, for example, named `condition`:

```
condition
control
control
disease1
disease2
...
```

And then change idents manually.

```r
Idents(obj) <- obj$condition
```
 Now, how to create this column is another question. Here is an example:

```r
# extract starting alphabets from current idents to create new vector
x <- stringr::str_extract(Idents(obj),"^[A-Za-z]+")
# rename to new values
x <- dplyr::recode(x,`F`="Control",`HD`="Disease1",`ILD`="Disease2")
# assign rownames
names(x) <- rownames(obj)
# add new vector to seurat metadata
obj$condition <- x
```
